### PR TITLE
[OpenThread] report RSSI for openthread

### PR DESCRIPTION
**Pull request by Hyungsin**
I really appreciate the openthread port #7082, which is @jia200x's job. 
I'm building on #7082 now and this PR deals with one of some issues in #7082 . 

OpenThread uses RSSI [dBm] as the link quality indicator, but the current OpenThread port does not provide a correct RSSI value. This PR provides RSSI for openthread and it supports SAMR21.

### CODE REVIEW

File: pkg/openthread/contrib/platform_radio.c [hunk 1]
```diff
@@ -138,8 +138,7 @@ void recv_pkt(otInstance *aInstance, netdev_t *dev)
     DEBUG("Openthread: Received pkt\n");
     netdev_ieee802154_rx_info_t rx_info;
     /* Read frame length from driver */
-    int len = dev->driver->recv(dev, NULL, 0, &rx_info);
-    Rssi = rx_info.rssi;
```
Comment (gebart) [inline pkg/openthread/contrib/platform_radio.c]: You should fix the Samr21 transceiver driver if it sets that wrong value in the rx_info struct, not add a radio specific workaround in one of the consumers.

File: pkg/openthread/contrib/platform_radio.c [hunk 2]
```diff
@@ -152,10 +151,17 @@ void recv_pkt(otInstance *aInstance, netdev_t *dev)
...
      * OpenThread do not use the data so we don't need to calculate FCS */
     sReceiveFrame.mLength = len + RADIO_IEEE802154_FCS_LEN;
-    sReceiveFrame.mPower = _get_power();
 
     /* Read received frame */
-    int res = dev->driver->recv(dev, (char *) sReceiveFrame.mPsdu, len, NULL);
+    int res = dev->driver->recv(dev, (char *) sReceiveFrame.mPsdu, len, &rx_info);
+
+    /* Conversion of raw RSSI value to dBm */
+#ifdef MODULE_AT86RF231 | MODULE_AT86RF233
```
Comment (gebart) [inline pkg/openthread/contrib/platform_radio.c]: This should not be necessary with a correct driver implementation.

**Comment by gebart** [comment]
NACK, fix the driver if the RSSI value is not correct, don't add driver specific workarounds in a device user.

**Comment by Hyungsin** [comment]
@gebart, just in case, do we have consensus that all radio drivers should report RSSI in [dBm]?

**Comment by gebart** [comment]
The documentation needs improvements, https://riot-os.org/api/structnetdev__radio__rx__info.html
It would make sense to specify it as dbm. 

If comparing RSSI between different packets from the same transceiver it should not even matter what the unit is as long as one number is greater than the other.

**Comment by gebart** [comment]
Also, why does OpenThread not use the LQI value as LQI?

**Comment by Hyungsin** [comment]
@gebart, please update documents then. It is confusing.

Especially, OpenThread requires that RSSI should be presented as dBm. 
It uses SNR [dBm] for link quality indicator, which is RSSI[dBm] - Noise[dBm]. 
It quantizes SNR to link quality value (0, 1, 2, or 3).  

So, this PR is good as it is?

**Comment by bergzand** [comment]
@Hyungsin did you see #6895? I want the same functionality as you there (conversion to dBm).

**Comment by Hyungsin** [comment]
@bergzand Oh, I didn't know. Good to know that thanks!

@gebart, based on it is good for the radio driver provides RSSI in dBm. But based on #6887, I don't think that all radio drivers can do that since some long range radios provide RSSI lower than -128dBm, which cannot be presented with 'int8_t' type. So it may also make sense that a radio driver just provides a register value itself with "uint8_t" and a driver user changes it to the dBm if she wants. How do you think about it?

**Comment by gebart** [comment]
@Hyungsin 
> @gebart, please update documents then. It is confusing.

I don't have time right now to work on this. Feel free to propose an improvement as a PR, it's all in the source tree https://github.com/RIOT-OS/RIOT/blob/master/drivers/include/net/netdev.h#L242-L250

> @gebart, it is good for the radio driver provides RSSI in dBm. To this end, we need to change the rssi variable type from 'uint8_t' to 'int8_t', which does not require a driver-specific fix but impacts the whole system. But based on #6887, I don't think that all radio drivers can support that since some long range radios provide RSSI lower than -128dBm, which cannot be presented with 'int8_t' type. So it may also make sense that a radio driver just provides a register value itself or whatever unit with "uint8_t" and a driver user changes it to the [dBm] if she wants. How do you think about it?

See my comment in https://github.com/RIOT-OS/RIOT/issues/6887#issuecomment-325275810

**Comment by Hyungsin** [comment]
@gebart, cool and thanks!

1. It seems that #6895 directly handles the radio rssi issue. IMHO, it would be good to let @bergzand keep working on #6895 reflecting your comment in #6887.  (uint8_t -> int16_t) @bergzand, can you do that?

2. Then, I think this PR can just focus on fixing OpenThread port under the assumption that a radio driver reports RSSI in dBm. Do you agree?

3. Maybe after #6887 is merged, I can work on updating the documentation :)

**Comment by gebart** [comment]
@Hyungsin looks like a good plan :+1:

**Comment by Hyungsin** [comment]
@gebart, alright. I fixed this PR under the assumption that the radio gives RSSI with dBm unit. 
Please check.

**Comment by Hyungsin** [comment]
@jia200x and @biboc, I think you guys can review this PR.

**Comment by jia200x** [comment]
Hi @Hyungsin 

Sure. I will try to find some time to review it.

**Comment by jia200x** [comment]
Looks good to me. Let's wait #6887 👍

**Comment by jia200x** [comment]
Sure. I will test it as soon as posible.

**Comment by jia200x** [comment]
Test on IoTLAB. Works OK for me!

**Comment by jia200x** [comment]
With the assumption that #6895 will be merged soon, this PR is ready to be merged in my opinion.

**Comment by Hyungsin** [comment]
Yeah, let's go forward. 
I'm mainly working on OpenThread/RIOT these days and there are many things to update given that openthread is being updated too!

**Comment by jia200x** [comment]
Murdock complains on PR needing squashing (long name?)

**Comment by Hyungsin** [comment]
@jia200x, I'm not familiar with the styling thing... Can you please let me know what this problem is?

**Comment by jia200x** [comment]
@Hyungsin I'm not sure why Murdock is complaining, there's only one commit.

@gebart do you know what's the problem?

**Comment by bergzand** [comment]
I think "fix" in the commit message is triggering the script: `dist/tools/pr_check/pr_check.sh`. It looks like it's trying to detect fixup commits by doing a grep for commit messages with "fix" in them

**Comment by jia200x** [comment]
@bergzand good catch! :) Thank you.

**Comment by Hyungsin** [comment]
@bergzand WoW. So do I need to change the commit message?

**Comment by jia200x** [comment]
@Hyungsin yes, that should do the trick

**Comment by Hyungsin** [comment]
Done. Thanks all!

**Comment by gebart** [comment]
You should prefix the commit message with the name of the module being changed, for example : "pkg/openthread: store RSSI value reported by radio driver for RX packets" or something

**Comment by Hyungsin** [comment]
@gebart, done. Thanks!

**Pull request merged**